### PR TITLE
[nrf noup] platform: nordic_nrf: nRF54L15 does not have UICR

### PR DESCRIPTION
**Pull request by MarkusLassila**
Do not attempt to lock the UICR.APPROTECT or UICR.SECUREAPPROTECT for nRF54L15. These registers do not exist.

TF-M provisioning with nRF54L15 will still result in compilation failure. As it should at this point.

### CODE REVIEW

**Review by tomi-font — CHANGES_REQUESTED** [review]
No `Change-Id` in noups.

File: platform/ext/target/nordic_nrf/common/core/target_cfg.c
Comment (tomi-font) [inline platform/ext/target/nordic_nrf/common/core/target_cfg.c:817]: How about checking e.g. for `NRF_UICR_S` instead?

Comment (tomi-font) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:817]: Or some register/define in particular under UICR (`APPROTECT`, `SECUREAPPROTECT`).

Comment (MarkusLassila) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:817]: I think that we should use the devices as we are using them elsewhere. This is maybe something that could be changed when we know how the nRF54L15 is going to look. We want it to lock debugger permanently, but it is not supported in 2.8.

Comment (tomi-font) [reply platform/ext/target/nordic_nrf/common/core/target_cfg.c:817]: though we are using `defined(NRF_APPROTECT)` here already :thinking:

**Comment by MarkusLassila** [comment]
> > TF-M provisioning with nRF54L15 will still result in compilation
> > failure. As it should at this point.
> 
> I don't see what's the link between this and what you're doing here?

nrf_provisioning.c checks that these UICR registers are set. This will still cause compilation error for nRF54L15.

**Comment by nika-nordic** [comment]
let me check this change with my PR: https://github.com/nrfconnect/sdk-zephyr/pull/2162 before merging this

**Comment by nika-nordic** [comment]
LGTM, but I will approve once CI on https://github.com/nrfconnect/sdk-nrf/pull/18370 is green.

Testing done locally went fine and @MarkusLassila took a look at that as well

**Comment by MarkusLassila** [comment]
@nordicjm: When the https://github.com/nrfconnect/sdk-zephyr/pull/2162 is backported to v2.8 branch, do we need to have a corresponding branch for this as well? I am not sure how that should be achieved.

**Comment by nordicjm** [comment]
> @nordicjm: When the [nrfconnect/sdk-zephyr#2162](https://github.com/nrfconnect/sdk-zephyr/pull/2162) is backported to v2.8 branch, do we need to have a corresponding branch for this as well? I am not sure how that should be achieved.

You should have a branch already where the tag for rc2 was made from

**Comment by MarkusLassila** [comment]
> > @nordicjm: When the [nrfconnect/sdk-zephyr#2162](https://github.com/nrfconnect/sdk-zephyr/pull/2162) is backported to v2.8 branch, do we need to have a corresponding branch for this as well? I am not sure how that should be achieved.
> 
> You should have a branch already where the tag for rc2 was made from

So, likely this one: v2.1.1-ncs1-rc1

**Review by nika-nordic — APPROVED** [review]
sdk-nrf PR is green with APPROTECT changes

**Review by Vge0rge — APPROVED** [review]

**Comment by rlubos** [comment]
@MarkusLassila @nika-nordic This will need to be manually backported

**Comment by MarkusLassila** [comment]
> @MarkusLassila @nika-nordic This will need to be manually backported

@rlubos @nika-nordic: 2.8 available in here: https://github.com/nrfconnect/sdk-trusted-firmware-m/pull/174

**Pull request merged**
